crates/tx5-connection/src/webrtc/libdatachannel.rs

```diff
@@ -0,0 +1,382 @@
+            if event_enabled!(target: "datachannel", Level::WARN) {
+                datachannel::configure_logging(Level::WARN);
+                return;
+            }
+            if event_enabled!(target: "datachannel", Level::ERROR) {
+                datachannel::configure_logging(Level::ERROR);
+                return;
+            }
+        });
+
+        let (mut cmd_send, cmd_recv) = CloseSend::sized_channel(1024);
+        let (mut evt_send, evt_recv) = CloseSend::sized_channel(1024);
+
+        let task = tokio::task::spawn(task(
+            is_polite,
+            config,
+            send_buffer,
+            evt_send.clone(),
+            cmd_send.clone(),
+            cmd_recv,
+        ));
+
+        cmd_send.set_close_on_drop(true);
+        evt_send.set_close_on_drop(true);
+
+        let this: DynWebrtc = Arc::new(Self {
+            cmd_send,
+            _task: AbortTask(task),
+            _evt_send: evt_send,
+        });
+
+        (this, evt_recv)
+    }
+}
+
+impl super::Webrtc for Webrtc {
+    fn in_offer(&self, offer: Vec<u8>) -> BoxFuture<'_, Result<()>> {
+        Box::pin(async move {
+            self.cmd_send
+                .send(Cmd::InOffer(offer))
+                .await
+                .map_err(|_| Error::other("closed"))
+        })
+    }
+
+    fn in_answer(&self, answer: Vec<u8>) -> BoxFuture<'_, Result<()>> {
+        Box::pin(async move {
+            self.cmd_send
+                .send(Cmd::InAnswer(answer))
+                .await
+                .map_err(|_| Error::other("closed"))
+        })
+    }
+
+    fn in_ice(&self, ice: Vec<u8>) -> BoxFuture<'_, Result<()>> {
+        Box::pin(async move {
+            self.cmd_send
+                .send(Cmd::InIce(ice))
+                .await
+                .map_err(|_| Error::other("closed"))
+        })
+    }
+
+    fn message(&self, message: Vec<u8>) -> BoxFuture<'_, Result<()>> {
+        Box::pin(async move {
+            let (s, r) = tokio::sync::oneshot::channel();
+            self.cmd_send
+                .send(Cmd::SendMessage(message, s))
+                .await
+                .map_err(|_| Error::other("closed"))?;
+            let _ = r.await;
+            Ok(())
+        })
+    }
+}
+
+async fn task(
+    is_polite: bool,
+    config: Vec<u8>,
+    send_buffer: usize,
+    evt_send: CloseSend<WebrtcEvt>,
+    cmd_send: CloseSend<Cmd>,
+    cmd_recv: CloseRecv<Cmd>,
+) {
+    if let Err(err) =
+        task_err(is_polite, config, send_buffer, evt_send, cmd_send, cmd_recv)
+            .await
+    {
+        tracing::warn!(?err, "webrtc task error");
+    }
+}
+
+async fn task_err(
+    is_polite: bool,
+    config: Vec<u8>,
+    send_buffer: usize,
+    mut evt_send: CloseSend<WebrtcEvt>,
+    cmd_send: CloseSend<Cmd>,
+    mut cmd_recv: CloseRecv<Cmd>,
+) -> Result<()> {
+    evt_send.set_close_on_drop(true);
+
+    #[derive(Debug, Default, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct U {
```

Comment:
U for something that contains urls, and C for something that contains ice servers? connections?

Reply:
Hehe, something like that. Since these structs are only used inline, and not outside this function I wasn't too concerned about the names, but if you think it's confusing I can try to come up with something : )

Reply:
Ah, fine by me as it is.